File: DevExpress-Demo/VB/RichEditDemo.Wpf/Modules/DocumentProtection/UserService.vb
Imports System.Collections.Generic
Imports DevExpress.XtraRichEdit.Services

Namespace RichEditDemo

    Public Class UserService
        Implements IUserListService

        Private ReadOnly usersField As List(Of String) = New List(Of String)()

        Public ReadOnly Property Users As List(Of String)
            Get
                Return usersField
            End Get
        End Property

        Private Function GetUsers() As IList(Of String) Implements IUserListService.GetUsers
            Return Users
        End Function

        Public Sub Update(ByVal userList As List(Of String))
            usersField.Clear()
            usersField.AddRange(userList)
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/RichEditDemo.Wpf/Modules/DocumentProtection/UserService.cs
using System;
using System.Collections.Generic;
using DevExpress.XtraRichEdit.Services;

namespace RichEditDemo {
    public class UserService : IUserListService {
        readonly List<string> users = new List<string>();

        public List<string> Users { get { return users; } }

        IList<string> IUserListService.GetUsers() {
            return Users;
        }
        public void Update(List<String> userList) {
            this.users.Clear();
            this.users.AddRange(userList);
        }
    }
}
